code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -38,6 +38,51 @@
 		if((m != R.module_state_1) && (m != R.module_state_2) && (m != R.module_state_3))
 			. += m
 
+/obj/item/weapon/robot_module/proc/add_module(var/obj/item/I)
+	if(istype(I, /obj/item/stack))
+		var/obj/item/stack/S = I
+
+		if(is_type_in_list(S, list(/obj/item/stack/sheet/metal, /obj/item/stack/rods, /obj/item/stack/tile/plasteel)))
+			var/datum/robot_energy_storage/metal/metstore = locate(/datum/robot_energy_storage/metal) in storages
+			if(!metstore)
+				metstore = new /datum/robot_energy_storage/metal(src)
+
+			if(S.materials[MAT_METAL])
+				S.cost = S.materials[MAT_METAL] / 4

```

Comment:
@duncathan so in fancy, you're just saying if the multiplication form of the divide (the reciprocal of the divide) doesn't leave .0000000 go ahead and use it instead of the divide, ya?